clients/datadog/datadog_test.go

```diff
@@ -6,6 +6,20 @@ import (
 	"testing"
 )
 
+func TestDataDogClient_CountWithValue(t *testing.T) {
+	client := NewDatadogClient(DataDogConfig{
+		DD_AGENT_HOST: "localhost",
+		DD_AGENT_PORT: 8125,
+	})
+
+	t.Run("Should count with custom value", func(t *testing.T) {
+		a := assert.New(t)
+
+		err := client.CountWithValue("test_count", 5, map[string]string{"test": "test"}, 1.0)
+		a.NoError(err)
```

Comment:
```suggestion
		a := assert.New(t)
		ctrl := gomock.NewController(t)
		defer ctrl.Finish()

		mockClient := mocks.NewMockClientInterface(ctrl)
		client.Client = mockClient

		mockClient.EXPECT().Count(
			"test_count",           // metric name
			int64(5),               // value
			[]string{"test:test"},  // tags
			1.0,                    // rate
		).Return(nil)

		err := client.CountWithValue("test_count", 5, map[string]string{"test": "test"}, 1.0)
		a.NoError(err)
```

i guess, we could extend the test